src/tests/ftest/server/metadata.py

```diff
@@ -252,29 +289,43 @@ def test_metadata_fillup(self):
                     sequential_fail_counter += 1
                 if sequential_fail_counter >= sequential_fail_max:
                     self.log.info(
-                        "Phase 3: container %d - %d/%d sequential no space "
+                        "Phase 1: container %d - %d/%d sequential no space "
                         "container create errors", sequential_fail_counter,
                         sequential_fail_max, loop)
                     break
 
                 if status and in_failure:
                     self.log.info(
-                        "Phase 3: container: %d - no space -> available "
+                        "Phase 1: container: %d - no space -> available "
                         "transition, sequential no space failures: %d",
                         loop, sequential_fail_counter)
                     in_failure = False
                 elif not status and not in_failure:
                     self.log.info(
-                        "Phase 3: container: %d - available -> no space "
+                        "Phase 1: container: %d - available -> no space "
                         "transition, sequential no space failures: %d",
                         loop, sequential_fail_counter)
                     in_failure = True
 
             except TestFail as error:
                 self.log.error(str(error))
-                self.fail("Phase 3: fail (unexpected container create error)")
+                self.fail("Phase 1: fail (unexpected container create error)")
+        if len(self.container) >= self.CREATED_CONTAINERS_LIMIT:
+            self.log.error("Phase 1: Created too many containers: %d > %d", len(self.container),
+                           self.CREATED_CONTAINERS_LIMIT)
+            self.fail("Phase 1: Created too many containers")
+        if len(self.container) < self.CREATED_CONTAINERS_MIN:
+            self.log.info("Phase 1: Created too few containers: %d < %d", len(self.container),
+                          self.CREATED_CONTAINERS_MIN)
+            self.fail("Phase 1: Created too few containers")
         self.log.info(
-            "Phase 3: passed (created %d / %d containers)", len(self.container), loop)
+            "Phase 1: passed (created %d / %d containers)", len(self.container), loop)
+
+        # Phase 2 clean up some containers (expected to succeed)
+        self.log.info("Phase 2: Cleaning up 500 containers (expected to work)")
+        if not self.destroy_n_containers(500):
+            self.fail("Phase 2: fail (unexpected container destroy error)")
```

Comment:
Does this only destroy 500 to speed up the test?

Reply:
Yes indeed. It was a realization I made while considering the impact of changing the test to use the default rdb size, and recognizing that the point of the destroys is to make sure that destroys can occur when rdb is full. So the 500 containers approach here ensures what the test intended, while also trying to conserve execution time.

Reply:
Good decision! I think it's good to think of cases like this